tests/cpptests/test_cpp_forkgc.cpp

```diff
@@ -136,20 +109,81 @@ static InvertedIndex *getTagInvidx(RedisSearchCtx *sctx, const char *field,
   return iv;
 }
 
-static std::string numToDocid(unsigned id) {
-  char buf[1024];
-  sprintf(buf, "doc%u", id);
-  return std::string(buf);
+class FGCTestTag : public FGCTest {
+protected:
+  const char *tag_field_name = "f1";
+
+  void SetUp() override {
+    FGCTest::SetUp();
+    RediSearch_CreateTagField(ism, "f1");
+  }
+};
+
+class FGCTestNumeric : public FGCTest {
+protected:
+  const char *numeric_field_name = "n";
+
+  void SetUp() override {
+    FGCTest::SetUp();
+    RediSearch_CreateNumericField(this->ism, numeric_field_name);
+  }
+};
+
+TEST_F(FGCTestNumeric, testNumeric) {
```

Comment:
this one seems very similar to test_cpp_range.cpp::RangeIndexTest::testNumericTreeMemory
anything I miss?

Reply:
They are similar, but `testNumericTreeMemory` purpose is to test that the new member of the tree (`invertedIndexesSize`) correctly captures the sum of all its ranges memory, which is updated both in insertions and in deletions(gc), by summing all ranges in the tree and comparing the sum to `invertedIndexesSize`.
`testNumeric` tests that the gc correctly updates the memory of the ranges. It relies on the correctness of `invertedIndexesSize` so it can avoid iterating the tree in its validations